src/models/step_arguments/data_table.js

```diff
@@ -1,5 +1,31 @@
 import _ from 'lodash'
 
+const truths = ['true', 'TRUE', 'True', 'yes', 'YES', 'Yes', 'y', 'Y', '1']
+const trueFalse = (v) => !!~truths.indexOf(v)
+const asIs = (v) => v
+const list = (v) => v.split(',')
+const date = (v) => new Date(v)
+
+const types = {
+  'string':  asIs,
+  'str':     asIs,
+  '':        asIs,
+  'int':     parseInt,
+  'integer': parseInt,
+  'double':  parseInt,
+  'float':   parseFloat,
+  'number':  parseFloat,
+  'bool':    trueFalse,
+  'boolean': trueFalse,
+  'y/n':     trueFalse,
+  'bit':     trueFalse,
+  date,
+  'datetime':date,
+  'array':   list,
+  list,
+  'json':    JSON.parse
```

Comment:
the proposed data-structure is introducing complexity instead of keeping things flat and simple.

Instead of a N(1) operation of entering a hash, the proposed structure now implies a nested loop search. 
why....

How bad is it needed for you, @charlierudolph ?

If it's crucial - I can start with the proposed structure and flatten it on load time